helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -176,12 +176,18 @@ private int run0() throws InterruptedException, DockerException {
             .build().newScheduler();
 
         while (!healthChecker.get().check(containerId)) {
-          // TODO(negz): Why aren't we using getContainerInfo() here?
-          final ContainerState state = docker.inspectContainer(containerId).state();
+          final ContainerState state = getContainerState(containerId);
+          if (state == null) {
+            final String err = "container " + containerId + "was not found during health checking, "
+                + "or has no State object";
+            log.warn(err);
+            throw new RuntimeException(err);
+          }
           if (!state.running()) {
-            log.warn("container exited during health checking: {}: {}: {}",
-                     config, containerId, state.exitCode());
-            throw new RuntimeException("container exited during health checking");
+            final String err = "container " + containerId + "exited during health checking. "
```

Comment:
This needs an additional space: `" exited during"...` otherwise you'll get the `containerId` glued to it.